### DISCUSSION TITLE
Cyberwarfare

### DISCUSSION THREAD

**Discussion by aidan-bowman** [Foundation-19/Foundation-19#781] [Design]
As many of you know, the world of SCP involves a large amount of hacking and cybersecurity, not just in the standard way that most institutions deal with, but also in a much stranger matter. Many anomalies interact with, if not fully reside on, networks and electronic systems, and as such are unable to be fully implemented without a robust system of mechanics rivalling that of physical combat. However, most SS13 servers have the absolute bare minimum when it comes to computer science, and our codebase does no better. In fact, our current system does even worse, with many features poorly functioning if not completely disabled. How can we expect to implement SCPs like 079, or groups like the Church of the Broken God, when players can't even run a program on a computer remotely?

We need to design and implement a functioning system to allow for this. As such, I wrote up this proposal.

### Goals

- Provide a complex internet "ecosystem" that allows for players to exist purely as a networking entity.
- Integrate said network to affect and control the real world
- Allow for fun and complex interactions between players, allowing for interesting situations such as a CI attempting to spread and cultivate a virus to break the network, and battling the system AI for control.

### Overview

Due to the large amount of electronic-interacting roles and network-only entities, there would have to be a world system specifically for the cyberworld. This would look less like the regular map that human players interact with, and more like a node-based system, where each node represents a server, console, computer, or tablet. Since each node is an electronic device, how a player would use and spread across nodes would be obvious. For example, an RnD server would allow you to control the data related to research, and could be used to passively spread to computers accessing RnD.

There would be 2 main categories of entities that reside on this node-based network:

1. Large forces that would control a large swath of nodes (system AI, 079, cyberattacks by other factions). For them, the cyberworld would be similar to many tabletop wargames, except for the real time aspect. They would have resources such as computing time, and use this for combatting other players, spreading across nodes, and performing actions.
2. Smaller forces that exist and transfer across single nodes (players trying to hack shit, smaller worms, AICs). For them, the cyberworld would be closer to an RPG game, where they trawl across the network and use the node they're on to perform actions and gain resources.

**Comment by Kaga-404** [comment]
With regards to this, consider looking into the incomplete PR for re-implementing terminal commands to computers. That can be a good basis for what can be done with computers, even by non-cyber participants, and then expanded in the future into something on these lines in a gameified cyberworld for actual AICs/computer-residing anomalies.

**Reply by aidan-bowman** [reply]
I took a look at said PR (you're referring to #619, right?) and I completely agree. We could first implement inter-computer interaction with terminal commands, and then gradually expand that to more built-in programs before we implement forms of actual cybersecurity. Plus, by adding terminal commands we can pave the way for more specialized systems that could increase the power and utility of modular computers (something we need to make a cyberworld system useful).

**Comment by TheTitanicEngineer** [comment]
The concept of having more interaction on systems similar to SCiPnet seems great, but I believe that expanding it outside the site to end up becoming a tabletop wargame goes off-target. I believe the idea could be used with the current SCiPnet system, which has some basic features like DoS attacks, etc; what could be done is to expand it further to allow more types of network hardware and add functionalities to the SCiPnet.

We could work with things like adding an external TComms relay, which could be targeted to prevent external communication(requesting ERT/MTF, fax, etc.) while not disrupting the facility's internal network.

**Reply by aidan-bowman** [reply]
I disagree. I think that adding in more electronic interactions would greatly benefit the theme and gameplay. Many skips interact in some way with computer systems (for example, 079), and without anything to support that interaction we couldn't implement said skips. As for the "regular player", this could allow for all sorts of action. For example, a traitor could use a tablet to remotely access door systems, but would have to either stay hidden from or fight off AICs and system administrators. Engineers might be tasked with repairing and maintaining the network against less intelligent threats (for example, regular viruses that people bring on-site accidentally). Bored office workers might find a way to hack into financial systems so they can get free food from vending machines.

**Reply by TheTitanicEngineer** [reply]
But then we would end up separating the game into facility gameplay and tabletop wargames, which doesn't end nicely and would likely be ignored most part of the time, whether those interactions but sticking to SCiPnet. There's definitely the possibility for additions on the network features that affect the facility, but adding stuff that has little to no effect on the general facility just ends up creating a second game.
SCP-079 is never connected to the internet due to it's containment procedures. Adding such off-site interaction is unlikely to get used in the end.
If we went and added a 'cyberworld' we would go off the site's purpose, it's not possible for a single site to handle all the operations of the foundation.

 The traitor example you gave is somewhat already implemented via the access decryptor(remotely controlling doors is not practical), and can be worked on even more.
